Extensions/Physics2Behavior/utils.ts

```diff
@@ -0,0 +1,25 @@
+namespace gdjs {
+  export namespace physics2 {
+    export const computeCurrentContactsFromStartedAndEndedContacts = <T>(
+      current: Array<T>,
+      started: Array<T>,
+      ended: Array<T>
+    ): void => {
+      const endedDuplicate = [...ended];
+      started.forEach((startedItem) => {
+        const isAlsoEndedIndex = endedDuplicate.indexOf(startedItem);
+        if (isAlsoEndedIndex !== -1) {
+          endedDuplicate.splice(isAlsoEndedIndex, 1);
+        } else {
+          current.push(startedItem);
+        }
+      });
+      endedDuplicate.forEach((endedItem) => {
+        const index = current.indexOf(endedItem);
+        if (index !== -1) {
+          current.splice(index, 1);
+        }
+      });
+    };
+  }
```

Comment:
I think that this method could be much simpler.

End events are eaten by a following start event.
```
[end start]
start [end start]
start [end start] end
[end start] end
```
This means that a `start` is always before a `end`. It could be 2 simple steps:
- add started contacts
- remove ended contacts

Reply:
I'll dig into it